packages/astro/src/types/public/config.ts

```diff
@@ -1699,9 +1699,46 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 		 * To use this feature with the Astro VS Code extension, you must also enable the `astro.content-intellisense` option in your VS Code settings. For editors using the Astro language server directly, pass the `contentIntellisense: true` initialization parameter to enable this feature.
 		 */
 		contentIntellisense?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.responsiveImages
+		 * @type {boolean}
+		 * @default `undefined`
+		 * @version 5.0.0
+		 * @description
+		 *
+		 * Enables and configures automatic responsive image generation for images in your project. Set to `true` or an object with configuration options.
+		 * When enabled it will not transform any images unless you pass the `layout` option to the image component, or set the `layout` option in the `responsiveImages` configuration.
+		 *
+		 * ```js
+		 * {
+		 *  experimental: {
+		 * 		responsiveImages: {
+		 * 			layout: 'responsive',
+		 * 		},
+		 * }
+		 * ```
+		 *
+		 */
+
+		responsiveImages?:
+			| boolean
+			| {
+					/**
+					 * @docs
+					 * @name responsiveImages.layout
+					 * @default `undefined`
+					 * @description
+					 * The default layout type for responsive images. Can be overridden by the `layout` prop on the image component.
```

Comment:
Do we have a type heading for listing out the possibilities? Noting that with this as is in docs, the possible values won't be shown. I think we should have each of the options listed out, with a quickie one-liner what they mean. Experimental docs, so doesn't have to be totally fleshed out yet!